### DISCUSSION TITLE
Server Sent Events

### DISCUSSION THREAD

**Discussion by kikmak42** [activej/activej#309] [Q&A] [answered]
Does ActiveJ supports Server Sent Events (SSE)? 
If it does support it, how to use it?

**Comment by eduard-vasinskyi** [comment, marked as answer]
Hello, @kikmak42

There is currently no API for Server Sent Events. However, you can easily achieve this functionality by utilizing the [CSP](https://activej.io/async-io/csp) module.

Just create a `ChannelSupplier` of events, encode each event as a separate `ByteBuf` and send it as a body stream of an HTTP response. And don't forget to add the header `Content-Type: text/event-stream`.
Here is a [demo snippet](https://gist.github.com/eduard-vasinskyi/c46d8fc62baca55a5def5cfc6099525e) that illustrates how it can be done: 

Each second, the server sends an event that contains the current time.
If you have more questions, please feel free to ask.
